doc/supported_platforms.md

```diff
@@ -0,0 +1,18 @@
+# Supported Host Platforms
+
+`vfkit` is a macOS-only virtualization tool that uses Apple's Virtualization framework to manage virtual machines.
+
+## Host Requirements
+
+The project aims to support the most recent macOS major version at all times, along with the two previous major versions. Support for the oldest supported version will be dropped three years after its release date or when Apple discontinues support, whichever occurs first.
+
+**macOS 14.0 or later** is recommended and fully supported. CI tested on:
+- macOS 14 (Sonoma) - CI tested on Apple Silicon
```

Comment:
We don’t really test Apple silicon in CI, apart from some non-virt unit tests, which I would not call "testing" :)
Imo, we can drop all explicit versions from this file, this way we don’t need to update it next year when 27 is released.